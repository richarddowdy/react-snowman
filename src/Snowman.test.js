import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";

// smoke test
it("renders without crashing", function(){
  render(<Snowman />);
});


// snapshot test
it("matches snapshot", function(){
  const { asFragment } = render(<Snowman />);
  expect(asFragment()).toMatchSnapshot();
});


it("image changes to image 1 after first incorrect guess", function(){
  const { queryByAltText, queryByTestId } = render(<Snowman />)

  const xButton = queryByTestId("x");
  fireEvent.click(xButton);

  expect(queryByAltText("Image 1")).toBeInTheDocument();
})


it("image stays same and letter revealed on correct guess", function(){
  const { queryByAltText, queryByTestId } = render(<Snowman />)

  const aButton = queryByTestId("a");
  fireEvent.click(aButton);

  expect(queryByTestId("guessedWord")).toBeInTheDocument("a____");
  expect(queryByAltText("Image 0")).toBeInTheDocument();
})

it("handle end of game win/loss", function(){
  const { queryByAltText, queryByTestId } = render(<Snowman />)

  const xButton = queryByTestId("x");
  fireEvent.click(xButton);
  const yButton = queryByTestId("y");
  fireEvent.click(yButton);
  const zButton = queryByTestId("z");
  fireEvent.click(zButton);
  const uButton = queryByTestId("u");
  fireEvent.click(uButton);
  const tButton = queryByTestId("t");
  fireEvent.click(tButton);
  // const vButton = queryByTestId("v");
  // fireEvent.click(vButton);
 

  expect(queryByTestId("lose")).toBeInTheDocument();
})



