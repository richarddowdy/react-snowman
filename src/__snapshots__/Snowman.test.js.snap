// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Snowman"
  >
    <img
      alt="Image 0"
      data-testid="snowmanImage"
      src="0.png"
    />
    <p>
      Number Wrong: 0
    </p>
    <p
      class="Snowman-word"
      data-testid="guessedWord"
    >
      _____
    </p>
    <p
      data-testid="buttons"
    >
      <button
        data-testid="a"
        value="a"
      >
        a
      </button>
      <button
        data-testid="b"
        value="b"
      >
        b
      </button>
      <button
        data-testid="c"
        value="c"
      >
        c
      </button>
      <button
        data-testid="d"
        value="d"
      >
        d
      </button>
      <button
        data-testid="e"
        value="e"
      >
        e
      </button>
      <button
        data-testid="f"
        value="f"
      >
        f
      </button>
      <button
        data-testid="g"
        value="g"
      >
        g
      </button>
      <button
        data-testid="h"
        value="h"
      >
        h
      </button>
      <button
        data-testid="i"
        value="i"
      >
        i
      </button>
      <button
        data-testid="j"
        value="j"
      >
        j
      </button>
      <button
        data-testid="k"
        value="k"
      >
        k
      </button>
      <button
        data-testid="l"
        value="l"
      >
        l
      </button>
      <button
        data-testid="m"
        value="m"
      >
        m
      </button>
      <button
        data-testid="n"
        value="n"
      >
        n
      </button>
      <button
        data-testid="o"
        value="o"
      >
        o
      </button>
      <button
        data-testid="p"
        value="p"
      >
        p
      </button>
      <button
        data-testid="q"
        value="q"
      >
        q
      </button>
      <button
        data-testid="r"
        value="r"
      >
        r
      </button>
      <button
        data-testid="s"
        value="s"
      >
        s
      </button>
      <button
        data-testid="t"
        value="t"
      >
        t
      </button>
      <button
        data-testid="u"
        value="u"
      >
        u
      </button>
      <button
        data-testid="v"
        value="v"
      >
        v
      </button>
      <button
        data-testid="w"
        value="w"
      >
        w
      </button>
      <button
        data-testid="x"
        value="x"
      >
        x
      </button>
      <button
        data-testid="y"
        value="y"
      >
        y
      </button>
      <button
        data-testid="z"
        value="z"
      >
        z
      </button>
    </p>
  </div>
</DocumentFragment>
`;
